### PR TITLE
Added Notification Handling Mechanism

### PR DESCRIPTION
**Pull request by svr8**
# Problem
- A mechanism to handle all bundled notifications was needed as described in issue [13](https://github.com/codeuino/Social-Platform-Donut/issues/13)

# Solution
- Publisher/Subscriber model makes it easy for various GUI components to interact with updates in notifications. This is implemented in `views/js/notify.js`.  We subscribe to the notifications when document is loaded. 
- A Bootstrap dropdown button is added that will open up a vertical list of notifications.

# Screenshots
Notification UI

### **Type of Change**
- [ ] Bug fix
- [ ] New Feature
- [x] Development of UI/UX prototypes
- [ ] Small refactor
- [ ] Change in Documentation

### **Checklist**
- [x] My code follow the same style as the codebase
- [ ] My Code changed required change in documentation
- [ ] I have updated the Readme accordingly
- [x] I made PR within **development branch only**

### CODE REVIEW

**Comment by iamgrawal** [comment]
Hey @svr8, this pull request is incomplete. It is not having any GUI for notifications. Please add that as well :smile:

**Comment by vaibhavdaren** [comment]
Good work 👍

**Comment by svr8** [comment]
Hi @iamgrawal, I was on my way to implement the GUI for this.. in ejs. However, following the discussion on slack, should I wait for the frontend to completely move to vue.js?

**Comment by iamgrawal** [comment]
@svr8 Oh yeah, you need to implement the frontend in vue.

**Comment by piyush97** [comment]
It'd be great if you could solve the conflicts.

**Comment by svr8** [comment]
Hi @iamgrawal , I'll need time to learn Vue framework. The notification handling, ie, the logical part has been implemented. I guess it won't be a problem to use it in future?

**Comment by iamgrawal** [comment]
Hey @svr8 , what's the estimated time you need to work on this issue?

**Comment by svr8** [comment]
@iamgrawal I will update this PR by 15th April.

**Comment by iamgrawal** [comment]
@svr8 I am assigning this issue to you, please make sure that you do it till 15th.

**Comment by svr8** [comment]
Hi @iamgrawal , I have got a hang of Vue.js. I am listening to notifications from server using socket. Just for prototype development, I have hardcoded 2 default notifications in `/views/js/notify.js`. Please review.

File: views/other-landing.ejs [hunk 1]
```diff
@@ -1,281 +1,307 @@
...
+                      <li class="nav-item">
+                        <a class="nav-link" href="/profile/setting"><%=use.fname %><%=use.lname%></a>
+                      </li>
+                      <li class="nav-item">
+                        <a class="nav-link" href="/profile/dashBoard">DashBoard</a>
+                      </li>
+                      <li class="nav-item">
+                        <a class="nav-link" href="/profile/submitProject">Publish</a>
+                      </li>
+                      <li class="nav-item">
```
Comment (svr8) [inline views/other-landing.ejs]: This is code for Notifications.

File: views/other-landing.ejs [hunk 2]
```diff
@@ -1,281 +1,307 @@
...
+           });
+          })
+        })
+</script>
+        <!-- Bootstrap tooltips -->
+        <script type="text/javascript" src="/vendor/bootstrap/js/popper.min.js"></script>
+        <!-- Bootstrap core JavaScript -->
+        <script type="text/javascript" src="/vendor/bootstrap/js/bootstrap.bundle.min.js"></script>
+  
+        <!-- Notification Handler -->
```
Comment (svr8) [inline views/other-landing.ejs]: Import for Vue.js handling.

**Comment by iamgrawal** [comment]
> Hi @iamgrawal , I have got a hang of Vue.js. I am listening to notifications from server using socket. Just for prototype development, I have hardcoded 2 default notifications in `/views/js/notify.js` Also, I have added a screenshot in PR description. Please review.

I am not able to understand what are you trying to say with "got a hang of Vue.js".

**Comment by svr8** [comment]
I meant, I learned the basics of VueJS, practised implementing some modules in Vue.

**Comment by prateekiiest** [comment]
@svr8  could you please resolve the conflicts here?

**Comment by svr8** [comment]
Oh, I forgot that this is .ejs file and ejs contributions aren't being accepted.

**Pull request closed without merging**